core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchange.java

```diff
@@ -254,6 +292,22 @@ private static boolean isSystemPartitioning(PartitioningHandle partitioning)
         return partitioning.getConnectorHandle() instanceof SystemPartitioningHandle;
```

Comment:
nit: `isSystemPartitioning` should be moved to `SystemPartitioningHandle` class.